python/vowpalwabbit/pyvw.py

```diff
@@ -237,14 +282,25 @@ def format_input(key, val):
         if arg_str is not None:
             l = [arg_str] + l
 
-        pylibvw.vw.__init__(self, " ".join(l))
+        pylibvw.vw.__init__(self, " ".join(l), partial_initialize)
 
         # check to see if native parser needs to run
         ext_file_args = ["d", "data", "passes"]
+        self.needs_parser = False
         if any(x in kw for x in ext_file_args):
+            self.needs_parser = True
+
+        # TODO: clean this up
```

Comment:
What is the clean up required?

Reply:
Mostly the workflow. This is a little weird and unintuitive, where you either complete the initialization or not based on some flag that you pass in. I'll clarify this comment